src/components/theme/Image/Image.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect, useCallback, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { getImageAttributes } from '@plone/volto/helpers';
+
+/**
+ * Image component
+ * @param {object | string} image - Plone image as object or url
+ * @param {string} imageField - (default: image) image field for scales URL
+ * @param {string} alt - Alternative text for image
+ * @param {string} className - CSS class attribute
+ * @param {string} containerClassName - CSS class attribute for picture element
+ * @param {string} floated - float left or right
+ * @param {string} size - actual width: thumb, small, medium or large
+ * @param {string} role - img role attribute
+ * @param {boolean} critical - whether to lazy load the image
+ * @param {number} maxSize - maximum size to render
+ * @param {boolean} useOriginal - whether to render original size
+ */
+const Image = ({
+  image,
+  imageField = 'image',
+  alt = '',
+  className,
+  containerClassName,
+  floated,
+  size,
+  role = 'img',
+  critical = false,
+  maxSize,
+  useOriginal = false,
+  ...imageProps
+}) => {
+  const { src, srcSet } = getImageAttributes(image, {
+    imageField,
+    maxSize,
+    useOriginal,
+  });
+  const imageRef = useRef();
+  const [srcset, setSrcset] = useState(
+    critical && srcSet ? srcSet.join(', ') : null,
+  );
+  const imageHasLoaded = imageRef?.current?.complete;
+
+  let pictureClassName = `volto-image${
+    containerClassName ? ` ${containerClassName}` : ''
+  }`;
+  if (floated) {
+    pictureClassName = `${pictureClassName} floated ${floated}`;
+  }
+  if (size) {
+    pictureClassName = `${pictureClassName} ${size}`;
+  }
+
+  const applySrcSet = useCallback(() => {
+    setSrcset(
+      srcSet
+        .filter(
+          (s) =>
+            parseInt(s.split(' ')[1].replace('w', ''), 10) <=
+            (imageRef?.current?.width ?? Infinity),
+        )
+        .join(', '),
+    );
+  }, [srcSet]);
+
+  useEffect(() => {
+    if ('IntersectionObserver' in window) {
+      const observer = new IntersectionObserver(
+        (entries) => {
+          setTimeout(() => {
+            if (
+              entries[0].isIntersecting === true &&
+              imageRef?.current?.complete &&
+              (!srcset || srcset?.split(', ')?.length < 2) &&
+              srcSet?.length > 0
+            ) {
+              applySrcSet();
+            }
+          }, 600);
+        },
+        { threshold: [0] },
+      );
+      observer.observe(imageRef.current);
+    } else if (srcSet?.length > 0) {
+      applySrcSet();
+    }
+  }, [imageRef, applySrcSet, imageHasLoaded, srcSet, srcset]);
+
+  return (
+    <>
+      <picture className={pictureClassName}>
+        {srcset?.length > 0 && <source srcSet={srcset} />}
+        <img
+          src={src}
+          alt={alt}
+          className={className}
+          role={role}
+          loading="lazy"
+          style={{ width: '100%', objectFit: 'cover' }}
```

Comment:
Why the default styles? These look quite opinionated, even if they can be overriden with the `imageProps` prop. Is there a reason for these two styles?

Reply:
If I remember it correctly, I set those inline to avoid layout shifts while loading CSS.